types/index.d.ts

```diff
@@ -0,0 +1,22 @@
+import Vue, { PluginFunction } from 'vue';
+
+declare class VueThinModal {
+  currentName: string;
+  push(name: string): void;
+  pop(): void;
+  replace(name: string): void;
+
+  static install: PluginFunction<never>;
+}
+
+export interface VueThinModalOptions {
+  autoMountPortal?: boolean;
+}
+
+declare module 'vue/types/vue' {
+  interface Vue {
+    readonly $modal: VueThinModal;
+  }
+}
+
+export default VueThinModal;
```

Comment:
Can we replace it with just an object having `install` method? Because the `install` function and `$modal` object is not related.